### PR TITLE
feature/UAR-495-add-user-submission-data-to-new-JSON-template

### PR DESCRIPTION
**Pull request by vgummadala**
comment clean-up

Redundants removed & formatting plug-in applied

Changes post styling using google-java-format plugin & changes to BeneficialOwnerStatementType

updated with FilingForDate & commented on Presenter contact details

### JIRA link

https://companieshouse.atlassian.net/browse/UAR-495

### Work checklist

- [X] Tests added where applicable
### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java [hunk 1]
```diff
@@ -3,29 +3,36 @@
...
-   @JsonCreator
-   public static BeneficialOwnersStatementType findByBeneficialOwnersStatementTypeString(String beneficialOwnersStatement) {
-       for (BeneficialOwnersStatementType type: values()) {
-           if(type.beneficialOwnersStatement.equalsIgnoreCase(beneficialOwnersStatement)) {
-               return type;
-           }
-       }
-       return null;
-   }
+  /* getValue() returns the selected value (for e.g. in Update Submission) from this enum list */
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java:34]: The formatting of this comment doesn't seem quite right and is not consistent with the other javadoc in the class

```suggestion
  /**
   * getValue() returns the selected value (for e.g. in Update Submission) from this enum list
   * @return String
   */
```

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    filingForDate.setDay((String.valueOf(filingDate.getDayOfMonth())));
+    filingForDate.setMonth((String.valueOf(filingDate.getMonth())));
+    filingForDate.setYear((String.valueOf(filingDate.getYear())));
+    updateSubmission.setFilingForDate(filingForDate);
+  }
+
+  public void populatePresenter(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    Presenter presenter = new Presenter();
+    presenter.setEmail(overseasEntitySubmissionDto.getPresenter().getEmail());
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:76]: Can `overseasEntitySubmissionDto.getPresenter()` be null?

**Review by RobertBromfield — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java [hunk 3]
```diff
@@ -136,8 +138,9 @@ private void setFilingApiData(
       publicPrivateDataCombiner.buildMergedBeneficialOwnerDataMap();
       publicPrivateDataCombiner.buildMergedManagingOfficerDataMap();
 
-      ApiLogger.infoContext("PublicPrivateDataCombiner",
-          publicPrivateDataCombiner.logCollatedData());
+      ApiLogger.infoContext("PublicPrivateDataCombiner", publicPrivateDataCombiner.logCollatedData());
+
+      UpdateSubmission updateSubmission = new PopulateUpdateSubmission(submissionDto, new UpdateSubmission()).populate();
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java:143]: What do we plan to do with this `updateSubmission` object?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    // https://companieshouse.atlassian.net/browse/UAR-461//
+    // TBD updateSubmission.setNoChangesInFilingPeriodStatement(); //
+
+    return updateSubmission;
+  }
+
+  public void populateDueDiligence(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    DueDiligenceDto dueDiligenceDto =
+        overseasEntitySubmissionDto.getDueDiligence(); // TBD .. which DueDiligence to use??//
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:49]: Please find this information out and remove the comment

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    DueDiligence dueDiligence = new DueDiligence();
+    dueDiligence.setDueDiligenceCorrespondenceAddress(dueDiligenceDto.getAddress());
+    dueDiligence.setAgentName(dueDiligenceDto.getName());
+    dueDiligence.setEmail(dueDiligenceDto.getEmail());
+    dueDiligence.setDateChecked(
+        dueDiligenceDto.getIdentityDate().toString()); // TBD .. check date format//
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:55]: We need to find this out too

**Review by n-r-b — CHANGES_REQUESTED** [review]
Left comments

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  private OverseasEntitySubmissionDto overseasEntitySubmissionDto;
+  private UpdateSubmission updateSubmission;
+
+  public PopulateUpdateSubmission(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    this.overseasEntitySubmissionDto = overseasEntitySubmissionDto;
+    this.updateSubmission = updateSubmission;
+  }
+
+  /* method populates values into UpdateSubmission for JSON output */
```
Comment (RobertBromfield) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:26]: Could you make this a Javadoc comment 
```suggestion
  /**
   * This method populates values into UpdateSubmission for JSON output.
   * @return A populated UpdateSubmission object.
   */
```

**Review by RobertBromfield — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java [hunk 7]
```diff
@@ -0,0 +1,210 @@
+package uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (n-r-b) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java:3]: Can you replace this with single imports for readability?

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+    managingOfficerCorporateDtoList.add(ManagingOfficerMock.getManagingOfficerCorporateDto());
+  }
+
+  {
+    trustDataDtoList.add(TrustMock.getTrustDataDto());
+  }
+
+  @Test
+  void testPopulateUpdateSubmissionJSON() throws Exception {
+    output = getJsonOutput();
```
Comment (n-r-b) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java:119]: It's not clear what exactly we are testing here. Usually we would create an instance of the class and then call the method we want to test. We'd mock the data being passed to the method and then verify the response is what we expect. Hiding all of this inside of 'getJsonOutput' is not helpful at all

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java [hunk 9]
```diff
@@ -3,29 +3,36 @@
...
+   * Json creator allows non-matching enum values to be interpreted by the method which determines
+   * them as null
+   *
+   * @param beneficialOwnersStatement
+   * @return
+   */
+  @JsonCreator
+  public static BeneficialOwnersStatementType findByBeneficialOwnersStatementTypeString(
+      String beneficialOwnersStatement) {
+    for (BeneficialOwnersStatementType type : values()) {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java:26]: It's not a type - a statement

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java [hunk 10]
```diff
@@ -3,29 +3,36 @@
...
-       for (BeneficialOwnersStatementType type: values()) {
-           if(type.beneficialOwnersStatement.equalsIgnoreCase(beneficialOwnersStatement)) {
-               return type;
-           }
-       }
-       return null;
-   }
+  /* getValue() returns the selected value (for e.g. in Update Submission) from this enum list */
+  public String getValue() {
+    return this.beneficialOwnersStatement.toLowerCase();
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java:36]: toLowerCase should not be needed. See values on lines 6-8.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+  public UpdateSubmission populate() {
+    // updateSubmission.type(); Type is already set in UserSubmission, has a default value of
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:28]: remove this comment

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    // "OE02";
+    updateSubmission.setUserSubmission(this.overseasEntitySubmissionDto);
+    populateDueDiligence(this.overseasEntitySubmissionDto, updateSubmission);
+    populatePresenter(this.overseasEntitySubmissionDto, this.updateSubmission);
+    updateSubmission.setBeneficialOwnerStatement(
+        this.overseasEntitySubmissionDto.getBeneficialOwnersStatement().getValue());
+
+    updateSubmission.setAnyBOsOrMOsAddedOrCeased(
+        this.overseasEntitySubmissionDto.getUpdate().isRegistrableBeneficialOwner());
+    // AC contact details is covered in --> overseasEntitySubmissionDto.getPresenter().getEmail();
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:38]: Please remove these comments too.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+    // TBD: NoChangesInFilingPeriodStatement will be done in
+    // https://companieshouse.atlassian.net/browse/UAR-461//
+    // TBD updateSubmission.setNoChangesInFilingPeriodStatement(); //
+
+    return updateSubmission;
+  }
+
+  public void populateDueDiligence(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    DueDiligenceDto dueDiligenceDto =
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:48]: There are two types of due dillegence in the overseasEntitySubmissionDto. Overseas and plain dd. Only one will be non null and that one should be used to populate the DD in the updateSubmission. The fields to add will vary slightly but I hope that is covered in the ODP.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+
+  public void populatePresenter(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    Presenter presenter = new Presenter();
+    presenter.setEmail(overseasEntitySubmissionDto.getPresenter().getEmail());
+    presenter.setName(overseasEntitySubmissionDto.getPresenter().getFullName());
+    updateSubmission.setPresenter(presenter);
+  }
+
+  public UpdateSubmission populate(
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:81]: why two public populate overloads? I think the one without args should be private? Could also pass the DTos in the constructor which might be better.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    this.overseasEntitySubmissionDto = overseasEntitySubmissionDto;
+    this.updateSubmission = updateSubmission;
+    return populate();
+  }
+
+  public OverseasEntitySubmissionDto getOverseasEntitySubmissionDto() {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:88]: why is this a property?

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+            overseasEntitySubmissionDto, new Errors(), LOGGING_CONTEXT);
+    verifyValidateFull();
+    verify(ownersAndOfficersDataBlockValidator, times(1))
+        .validateOwnersAndOfficersAgainstStatement(eq(overseasEntitySubmissionDto), any(), any());
+    PopulateUpdateSubmission populateUpdateSubmission =
+        new PopulateUpdateSubmission(overseasEntitySubmissionDto, new UpdateSubmission());
+    UpdateSubmission updateSubmission = populateUpdateSubmission.populate();
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    objectMapper = JsonMapper.builder().findAndAddModules().build(); //
```
Comment (AndreKramer7654) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java:170]: Unit tests should just be testing objects and not converting to JSON.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java (hunk 9 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java:26]: Think I miss-understood what this was. Use: Enum.ValueOf

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+    output = objectMapper.writeValueAsString(updateSubmission);
+    return output;
+  }
+
+  private void verifyValidateFull() {
```
Comment (AndreKramer7654) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java:175]: Not sure why we are checking these validators have been called. They are not part of this PR?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.DueDiligence;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.FilingForDate;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.Presenter;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.UpdateSubmission;
+
+/**
+ * This is a helper class for FilingsService when the filing is an update. The purpose of this class
+ * is to populate change submission values into UpdateSubmission.java with the ultimate gaol to
+ * produce a json output of change submissions. It uses OESubmissionDTO and UpdateDTO values to
+ * populate JSON template.
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:14]: This class is used to populates the update submission model. (not a JSON template).

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java (hunk 9 above)
Comment (vgummadala) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java]: This is not my code @AndreKramer7654 Andre . May be somebody had a different purpose of it

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 19]
```diff
@@ -0,0 +1,132 @@
...
+    DueDiligenceDto dueDiligenceDto =
+        overseasEntitySubmissionDto.getDueDiligence();
+    OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto = overseasEntitySubmissionDto.getOverseasEntityDueDiligence();
+
+    DueDiligence submissionDueDiligence = new DueDiligence();
+
+    submissionDueDiligence  = (overseasEntitySubmissionDto.getDueDiligence() != null) ? populateByDueDiligenceDto(submissionDueDiligence, dueDiligenceDto ) :
+             populateByOEDueDiligenceDto(submissionDueDiligence, overseasEntityDueDiligenceDto);
+
+    updateSubmission.setDueDiligence(submissionDueDiligence);
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:57]: remove blank line or lines (also space in " ) :"

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+
+
+  }
+
+
+
+  public DueDiligence populateByDueDiligenceDto(DueDiligence dueDiligence, DueDiligenceDto dueDiligenceDto) {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:64]: you missed out:
@Field("diligence")
    private String diligence;

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:64]: check all fields of DueDiligenceDto and  OverseasEntityDueDiligenceDto  have been converted.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 21]
```diff
@@ -0,0 +1,116 @@
...
+  private OverseasEntitySubmissionDto overseasEntitySubmissionDto;
+  private UpdateSubmission updateSubmission;
+
+  public PopulateUpdateSubmission(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    this.overseasEntitySubmissionDto = overseasEntitySubmissionDto;
+    this.updateSubmission = updateSubmission;
+  }
+
+  /** method populates values into UpdateSubmission for JSON output */
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:27]: /**
 * Method populates ...
 */

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+            ? populateByDueDiligenceDto(submissionDueDiligence, dueDiligenceDto)
+            : populateByOEDueDiligenceDto(submissionDueDiligence, overseasEntityDueDiligenceDto);
+    updateSubmission.setDueDiligence(submissionDueDiligence);
+
+  }
+
+  public DueDiligence populateByDueDiligenceDto(
+      DueDiligence dueDiligence, DueDiligenceDto dueDiligenceDto) {
+
+    dueDiligence.setDateChecked(dueDiligenceDto.getIdentityDate().toString());
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:59]: lines 59 to 64 and 74 to 79 could be split out to a common private method setDueDiligenceDetails()

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+    dueDiligence.setAmlRegistrationNumber(overseasEntityDueDiligenceDto.getAmlNumber());
+
+    return dueDiligence;
+  }
+
+  public void populateFilingForDate(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    FilingForDate filingForDate = new FilingForDate();
+    var filingDate = overseasEntitySubmissionDto.getUpdate().getFilingDate();
+    filingForDate.setDay((String.valueOf(filingDate.getDayOfMonth())));
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:89]: why an extra level of ()? and 2 lines following

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java [hunk 24]
```diff
@@ -0,0 +1,190 @@
...
+  {
+    managingOfficerCorporateDtoList.add(ManagingOfficerMock.getManagingOfficerCorporateDto());
+  }
+
+  {
+    trustDataDtoList.add(TrustMock.getTrustDataDto());
+  }
+
+  @Test
+  void testPopulateUpdateSubmissionForDueDiligence() throws Exception {
```
Comment (AndreKramer7654) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java:90]: maybe move this closer to OEDueDiligence test?

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java [hunk 25]
```diff
@@ -0,0 +1,190 @@
...
+    overseasEntitySubmissionDto.setDueDiligence(dueDiligenceDto);
+    overseasEntitySubmissionDto.setManagingOfficersIndividual(managingOfficerIndividualDtoList);
+    overseasEntitySubmissionDto.setManagingOfficersCorporate(managingOfficerCorporateDtoList);
+    overseasEntitySubmissionDto.setTrusts(trustDataDtoList);
+    overseasEntitySubmissionDto.setUpdate(updateDto);
+  }
+
+  private void setIsTrustWebEnabledFeatureFlag(boolean value) {
+    ReflectionTestUtils.setField(overseasEntitySubmissionDtoValidator, "isTrustWebEnabled", value);
+  }
```
Comment (AndreKramer7654) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java:183]: add newline

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java (hunk 9 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/BeneficialOwnersStatementType.java]: Yeah, it just got reformated so marked in PR.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 26]
```diff
@@ -0,0 +1,116 @@
...
+  }
+
+  public UpdateSubmission populate(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    this.overseasEntitySubmissionDto = overseasEntitySubmissionDto;
+    this.updateSubmission = updateSubmission;
+    return populate();
+  }
+
+  public void setOverseasEntitySubmissionDto(
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:112]: no need for this to be public. Test coverage is still very low and any getter/setters that are not used / tested are going to show up in that.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 27]
```diff
@@ -0,0 +1,116 @@
...
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto, UpdateSubmission updateSubmission) {
+    Presenter presenter = new Presenter();
+    if (overseasEntitySubmissionDto.getPresenter() != null) {
+      presenter.setEmail(overseasEntitySubmissionDto.getPresenter().getEmail());
+      presenter.setName(overseasEntitySubmissionDto.getPresenter().getFullName());
+    }
+    updateSubmission.setPresenter(presenter);
+  }
+
+  public UpdateSubmission populate(
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:105]: Can you remove this method as it's not used?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java [hunk 28]
```diff
@@ -0,0 +1,116 @@
...
+    dueDiligence.setDueDiligenceCorrespondenceAddress(overseasEntityDueDiligenceDto.getAddress());
+    dueDiligence.setSupervisoryBody(overseasEntityDueDiligenceDto.getSupervisoryName());
+    dueDiligence.setPartnerName(overseasEntityDueDiligenceDto.getPartnerName());
+    dueDiligence.setEmail(overseasEntityDueDiligenceDto.getEmail());
+    dueDiligence.setAmlRegistrationNumber(overseasEntityDueDiligenceDto.getAmlNumber());
+
+    return dueDiligence;
+  }
+
+  public void populateFilingForDate(
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:85]: Do we need to call this method or remove it? It's unused at the moment

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java [hunk 29]
```diff
@@ -0,0 +1,190 @@
...
+class PopulateUpdateSubmissionTest {
+
+  private final EntityNameDto entityNameDto = EntityNameMock.getEntityNameDto();
+  private final EntityDto entityDto = EntityMock.getEntityDto();
+  private final PresenterDto presenterDto = PresenterMock.getPresenterDto();
+  private final BeneficialOwnersStatementType beneficialOwnersStatement =
+      BeneficialOwnersStatementType.NONE_IDENTIFIED;
+  private final DueDiligenceDto dueDiligenceDto = DueDiligenceMock.getDueDiligenceDto();
+  private final UpdateDto updateDto = UpdateMock.getUpdateDto();
+  private final List<BeneficialOwnerIndividualDto> beneficialOwnerIndividualDtoList =
```
Comment (n-r-b) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/PopulateUpdateSubmissionTest.java:47]: Since these mock initialisations (lines 47-57 and lines 62-87)  are only used once by the `buildOverseasEntitySubmissionDto` method, we could move these into the function

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java (hunk 22 above)
Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:59]: Venu - this comment is not resolved? There's quite a bit of duplication here.

Comment (AndreKramer7654) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/update/PopulateUpdateSubmission.java:59]: You would need to create a shared interface so not possible.
